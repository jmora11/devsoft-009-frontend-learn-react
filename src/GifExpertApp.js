import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import PropTypes from 'prop-types'

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />

            <ol>
                {
                    categories.map(categorie =>
                        <GifGrid
                            key={ categorie }
                            category={ categorie }
                        />
                    )
                }
            </ol>
        </>
    )
}

GifExpertApp.propTypes = {

}

export default GifExpertApp