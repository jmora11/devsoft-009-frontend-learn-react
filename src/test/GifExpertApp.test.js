import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Gif Expert APP', () => {

    let wrapper = shallow( <GifExpertApp defaultCategories={ [] }/> );

    beforeEach( () => {
        wrapper = shallow( <GifExpertApp defaultCategories={ [] } /> );
    });

    test('debe hacer match con el snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        const categories = ['Naruto', 'Demon Slayer'];
        wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
});