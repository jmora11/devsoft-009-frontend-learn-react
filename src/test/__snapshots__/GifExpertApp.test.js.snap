// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Expert APP debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Gif Expert APP debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Demon Slayer"
      key="Demon Slayer"
    />
  </ol>
</Fragment>
`;
